js/statuspageApi.js

```diff
@@ -9,7 +11,40 @@ class StatuspageApi {
   }
 
   lastIncident() {
-    return this.data.incidents[0];
+    let chosenIncident = undefined;
+    let incidents = this.data.incidents[0];
+    let scheduledMaintenances;
+    let incidentUpdated;
+    let maintenanceUpdated;
+
+    // if scheduled_maintenances has a relevant hashtag, add to the scheduledMaintenances variable
+    // if not, the chosenIncident is automatically incidents
+    if (
+      this.data.scheduled_maintenances === undefined ||
+      !hashtagRegexp.test(
+        this.data.scheduled_maintenances[0].incident_updates[0].body
+      )
+    ) {
+      chosenIncident = incidents;
+    } else {
+      scheduledMaintenances = this.data.scheduled_maintenances[0];
+    }
+
+    if (chosenIncident === undefined) {
+      incidentUpdated = incidents.updated_at.slice(0,19)
+      maintenanceUpdated = scheduledMaintenances.updated_at.slice(0,19)
+
+      // check if they were updated at the same time
+      if (incidentUpdated === maintenanceUpdated) {
+        chosenIncident = incidents;
+      } else {
+        // else, check which one was the most recent
+        incidents.updated_at < scheduledMaintenances.updated_at
+          ? (chosenIncident = scheduledMaintenances)
+          : (chosenIncident = incidents);
```

Comment:
I think everything inside this big if block could be reduced to something like the following pseudocode:
```
if (maintenanceMoreRecentThanIncident())
  chosen = maintenance
else
  chosen = incident
```
And then just add the updated_at logic and variables to that `maintenanceMoreRecentThanIncident` method (feel free to pick another name here).